### PR TITLE
Add broadcasting push notifications

### PR DESCRIPTION
**Pull request by minhnc**
https://hellosundaymorning.atlassian.net/browse/DP-2679

### CODE REVIEW

File: notification/notification.go [hunk 1]
```diff
@@ -164,9 +169,17 @@ func (manager *AppMobileNotificationManager) store(ctx context.Context, token To
 
 }
 
-func (manager *AppMobileNotificationManager) findTokensByUser(userID string) (tokens []Token, err error) {
+// findTokens returns the tokens of userID
+// or all tokens if passing ""
+func (manager *AppMobileNotificationManager) findTokens(userID string) (tokens []Token, err error) {
```
Comment (youwithouto) [inline notification/notification.go:174]: For the param, better to use a pointer value to indicate that it's optional.

**Review by youwithouto — COMMENTED** [review]
Just one suggestion, otherwise looks good!

Also need to check if the perform is OK, as the original implementation was only dealing with a single notification and sync send is OK, but this update includes broadcast which may affect the performance.

**Comment by minhnc** [comment]
Performance is good with just ms response time.

**Review by youwithouto — APPROVED** [review]
Looks good!

**Pull request merged**